File: src/my_project_typescript/easy_problems_typescript/from101to150/countCommonWordsOneOccurrence.ts
function countWords(words1: string[], words2: string[]): number {

    let tempDict: any = {}
    

    for (let word of words1){
        if (!tempDict.hasOwnProperty(word)){
            tempDict[word] = [1,0]
        }
        else {
            tempDict[word][0] += 1
        }
    }
    for (let word of words2){
        if (!tempDict.hasOwnProperty(word)){
            tempDict[word] = [0,1]
        }
        else {
            tempDict[word][1] += 1
        }
    }



    let answer: number = 0

    for (let word in tempDict){
        
        if (tempDict[word][0] === 1 && tempDict[word][1] ===1){
            answer += 1
        }
    }

    return answer 

};

console.log(countWords(["leetcode","is","amazing","as","is"],  ["amazing","leetcode","is"]))

console.log(countWords(["b","bb","bbb"],  ["a","aa","aaa"]))

console.log(countWords(["a","ab"], ["a","a","a","ab"]))

File: src/my_project/easy_problems/from201to250/count_common_words_one_occurrence.py
from typing import List 

class Solution:
    def countWords(self, words1: List[str], words2: List[str]) -> int:
        
        temp_dict = {}
        
        for word in words1:
            if word not in temp_dict:
                temp_dict[word] = [1,0]
            else:
                temp_dict[word][0] += 1
        
        
        for word in words2:
            if word not in temp_dict:
                temp_dict[word] = [0,1]
            else: 
                temp_dict[word][1] += 1
                
        answer = 0
        for word in temp_dict:
            if temp_dict[word] == [1,1]:
                answer += 1
        return answer 
    
    
solution  = Solution()

print(solution.countWords(words1 = ["leetcode","is","amazing","as","is"], words2 = ["amazing","leetcode","is"]))

print(solution.countWords(words1 = ["b","bb","bbb"], words2 = ["a","aa","aaa"]))

print(solution.countWords(words1 = ["a","ab"], words2 = ["a","a","a","ab"]))
